scripts/pythondistdeps.py

```diff
@@ -334,83 +366,71 @@ def normalize_name(name):
+                            py_deps[name].append((spec.operator, spec.version))
             # Unused, for automatic sub-package generation based on 'extras' from egg/dist metadata
             # TODO: implement in rpm later, or...?
             if args.extras:
-                deps = dist.requires()
-                extras = dist.extras
-                print(extras)
-                for extra in extras:
+                print(dist.extras)
+                for extra in dist.extras:
                     print('%%package\textras-{}'.format(extra))
                     print('Summary:\t{} extra for {} python package'.format(extra, dist.key))
                     print('Group:\t\tDevelopment/Python')
-                    depsextras = dist.requires(extras=[extra])
-                    for dep in reversed(depsextras):
-                        if dep in deps:
-                            depsextras.remove(dep)
-                    deps = depsextras
-                    for dep in deps:
-                        for spec in dep.specs:
-                            if spec[0] == '!=':
-                                print('Conflicts:\t{} {} {}'.format(dep.key, '==', spec[1]))
+                    for dep in dist.requirements_for_extra(extra):
+                        for spec in dep.specifier:
+                            if spec.operator == '!=':
+                                print('Conflicts:\t{} {} {}'.format(dep.name, '==', spec.version))
                             else:
-                                print('Requires:\t{} {} {}'.format(dep.key, spec[0], spec[1]))
+                                print('Requires:\t{} {} {}'.format(dep.name, spec.operator, spec.version))
                     print('%%description\t{}'.format(extra))
                     print('{} extra for {} python package'.format(extra, dist.key))
                     print('%%files\t\textras-{}\n'.format(extra))
             if args.conflicts:
                 # Should we really add conflicts for extras?
                 # Creating a meta package per extra with recommends on, which has
                 # the requires/conflicts in stead might be a better solution...
-                for dep in dist.requires(extras=dist.extras):
-                    name = dep.key
-                    for spec in dep.specs:
-                        if spec[0] == '!=':
+                for dep in dist.requirements + dist.extra_requirements:
+                    for spec in dep.specifier:
+                        if spec.operator == '!=':
                             if name not in py_deps:
-                                py_deps[name] = []
-                            spec = ('==', spec[1])
+                                py_deps[dep.name] = []
+                            spec = ('==', spec.version)
                             if spec not in py_deps[name]:
-                                py_deps[name].append(spec)
+                                py_deps[dep.name].append(spec)
```

Comment:
`name` is not initialized here, and `dep.name` is another replacement for `dep.key` which is not consistent with the others.